src/agent/rustjail/src/container.rs

```diff
@@ -1297,6 +1303,31 @@ pub fn setup_child_logger(fd: RawFd, child_logger: Logger) -> tokio::task::JoinH
     })
 }
 
+/// Sets domainname for process (see
+/// [setdomainname(2)](https://man7.org/linux/man-pages/man2/setdomainname.2.html)).
+///
+/// unistd doesn't have a safe binding for this, so need to make our own.
+/// Pulled from https://github.com/containers/youki/blob/6d698da2c70ca7bfa71699495157acf3d9930657/crates/libcontainer/src/syscall/linux.rs#L350-L369
+/// with some small tweaks.
+fn set_domainname(name: &str) -> Result<()> {
+    let ptr = name.as_bytes().as_ptr() as *const c_char;
```

Comment:
nit: `cast` is nicer: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a9631617aa5c467bda3bd289574ae77f

Reply:
TIL, will change!

Reply:
Done